source/JsMaterialX/JsMaterialXCore/JsUtil.cpp

```diff
@@ -1,53 +1,38 @@
-#include "../helpers.h"
-#include <MaterialXCore/Node.h>
-#include <MaterialXCore/Util.h>
+#include "../vectorHelper.h"
 
-#include <MaterialXFormat/File.h>
-#include <iostream>
-#include <vector>
-#include <iterator>
-#include <map>
+#include <MaterialXCore/Util.h>
 
-#include <emscripten.h>
 #include <emscripten/bind.h>
 
 namespace ems = emscripten;
 namespace mx = MaterialX;
 
 extern "C"
 {
-    EMSCRIPTEN_BINDINGS(my_module)
+    EMSCRIPTEN_BINDINGS(util)
     {
+        ems::constant("EMPTY_STRING", mx::EMPTY_STRING);
+
         ems::function("getVersionString", &mx::getVersionString);
-        ems::function("createValidName", &mx::createValidName); // arg0 === {std::string}, arg1 === {unicode representing character}
-        ems::function("isValidName", &mx::isValidName);
-        ems::function("incrementName", &mx::incrementName);
 
-        // The following function throw: Cannot call {function name} due to unbound types: XXXXX
         ems::function("getVersionIntegers", ems::optional_override([]() {
-                     std::tuple<int, int, int> version = mx::getVersionIntegers();
-                     return arrayToVec((int *)&version, 3);
+                     auto version = mx::getVersionIntegers();
+                     return ems::val::array((int *)&version, (int *)&version + 3);
                  }));
 
-        ems::function("splitString", ems::optional_override([](std::string str, std::string sep) {
-                     const std::string &str1 = str;
-                     const std::string &sep2 = sep;
-                     return mx::splitString(str1, sep2);
-                 }));
+        ems::function("createValidName", ems::optional_override([](std::string name) {
```

Comment:
This function takes an optional parameter. Why not use `BIND_FUNC("createValidName", createValidName, std::string, char)` and remove the other override below?

Reply:
Because the optional param is a char and not a string and therefore emscripten expects you to provide a number and not something like `"b"`
This is not ideal as you can provide more characters, but the tradeoff for usability seems to be better this way